File: simple/csharp/move-method_isolate-switch_before.cs
class Order
{
  // ...

  public double calculateTotal()
  {
    double total = 0;
    foreach (Product product in getProducts())
    {
      total = product.quantity * product.price;
    }
    total = applyRegionalDiscounts(total);
    return total;
  }

  public double applyRegionalDiscounts(double total)
  {
    double result = total;
    switch (user.getCountry())
    {
      case "US": result *= 0.85; break;
      case "RU": result *= 0.75; break;
      case "CN": result *= 0.9; break;
      // ...
    }
    return result;
  }

File: simple/typescript/move-method_isolate-switch_before.ts
class Order {
  // ...

  calculateTotal(): number {
    let total = 0;
    for (let product : getProducts()) {
      total += product.quantity * product.price;
    }
    total = applyRegionalDiscounts(total);
    return total;
  }

  applyRegionalDiscounts(total: number): number {
    let result = total;
    switch (user.getCountry()) {
      case 'US': result *= 0.85; break;
      case 'RU': result *= 0.75; break;
      case 'CN': result *= 0.9; break;
      // ...
    }
    return result;
  }
}